plugins/rikki/heroeslounge/components/ViewApps.php

```diff
@@ -118,6 +120,15 @@ public function onAccept()
                     //a sloth should not participate in two different teams in the same season or tournament
                     if ($this->sloth->mayJoinTeam($team)) {
                         $this->sloth->teams()->add($team);
+
+                        $seasons = Season::all();
+                        foreach ($seasons as $season) {
+                            if ($season->reg_open && $season->teams->contains($team) && $season->free_agents->contains($this->sloth)) {
+                                $season->free_agents()->remove($this->sloth);
```

Comment:
Why did you change detach to remove?

Reply:
Since we use the adding equivalent when a sloth signs up for a season. I can use detach if you prefer though.

Reply:
Remove is fine if it works

Reply:
I tested it locally and it worked as expected